From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Mon, 21 Mar 2022 15:38:28 +0100
Subject: [PATCH] ARM: stm32mp: dts: lxa-tac: wire-in factory data

We have an EEPROM on each of the Base board (in the SIP) and on the
Powerboard. Both of which will contain a TLV blob. Describe this in the
device tree.

Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
---
 arch/arm/dts/stm32mp157c-lxa-tac.dts | 34 ++++++++++++++++++++++++++++++++--
 1 file changed, 32 insertions(+), 2 deletions(-)

diff --git a/arch/arm/dts/stm32mp157c-lxa-tac.dts b/arch/arm/dts/stm32mp157c-lxa-tac.dts
index 6301db2a337a..c5fd146edd47 100644
--- a/arch/arm/dts/stm32mp157c-lxa-tac.dts
+++ b/arch/arm/dts/stm32mp157c-lxa-tac.dts
@@ -16,6 +16,18 @@
 
 	aliases {
 		state = &state_emmc;
+		baseboard-factory-data = &baseboard_factory_data;
+		powerboard-factory-data = &powerboard_factory_data;
+	};
+
+	baseboard_factory_data: baseboard-factory-data {
+		compatible = "lxa,tlv-baseboard-v1";
+		backend = <&baseboard_tlv_backend>;
+	};
+
+	powerboard_factory_data: powerboard-factory-data {
+		compatible = "lxa,tlv-powerboard-v1";
+		backend = <&powerboard_tlv_backend>;
 	};
 
 	reserved-memory {
@@ -80,13 +92,31 @@
 	};
 };
 
-};
-
 &led_status_dut {
 	/* only turn on in Linux */
 	/delete-property/ linux,default-trigger;
 };
 
+&baseboard_eeprom {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	baseboard_tlv_backend: factory-data@0 {
+		reg = <0 0x400>;
+		label = "tlv";
+	};
+};
+
+&powerboard_eeprom {
+	#address-cells = <1>;
+	#size-cells = <1>;
+
+	powerboard_tlv_backend: factory-data@0 {
+		reg = <0 0x100>;
+		label = "tlv";
+	};
+};
+
 &sdmmc2 {
 	partitions {
 		compatible = "fixed-partitions";
